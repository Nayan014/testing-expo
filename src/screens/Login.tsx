import { View, Text, Image, TextInput, StyleSheet, Button } from "react-native";
import React, { useContext, useState } from "react";
import { AuthContext } from "../context/Auth";

const Login = () => {
  const [email, setEmail] = useState<string>("");
  const [password, setPassword] = useState<string>("");

  const authContext = useContext(AuthContext);

  return (
    <View
      style={{
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "white",
      }}
    >
      <Image
        source={{
          uri: "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/1200px-React-icon.svg.png",
        }}
        width={110}
        height={100}
      />
      <View style={{ width: "80%", marginTop: 40 }}>
        <TextInput
          onChangeText={(text) => setEmail(text)}
          value={email}
          placeholder="Please Enter Email"
          keyboardType="email-address"
          style={styles.input}
        />

        <TextInput
          onChangeText={(text) => setPassword(text)}
          value={password}
          placeholder="Please Enter Password"
          style={[styles.input, { marginTop: 14, marginBottom: 18 }]}
          secureTextEntry
        />

        <Button title="Login" onPress={() => authContext?.setAuth(true)} />
      </View>
    </View>
  );
};

export default Login;

const styles = StyleSheet.create({
  input: {
    borderColor: "gray",
    borderWidth: 0.5,
    borderRadius: 5,
  },
});
