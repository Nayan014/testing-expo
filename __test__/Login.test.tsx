import { fireEvent, render } from '@testing-library/react-native';
import Test from '../src/screens/Test';
import AuthProvider, { AuthContext } from '../src/context/Auth';
import Login from '../src/screens/Login';

const mockAuthContext = { auth: false, setAuth: jest.fn() };

const renderLoginScreen = () =>
  render(
    <AuthContext.Provider value={mockAuthContext}>
      <Login />
    </AuthContext.Provider>
  );

describe('Login Screen', () => {
  it('Render screen', () => {
    const { getByPlaceholderText, getByText, getByTestId } =
      renderLoginScreen();

    expect(getByPlaceholderText('Please Enter Email')).toBeTruthy();
    expect(getByPlaceholderText('Please Enter Password')).toBeTruthy();
    expect(getByText('Login')).toBeTruthy();
    expect(getByTestId('logo')).toBeTruthy();
  });

  it('render and match snapshot', () => {
    const tree = renderLoginScreen().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Use shoud able to enter email and password', () => {
    const { getByPlaceholderText } = renderLoginScreen();

    const emailInput = getByPlaceholderText('Please Enter Email');
    const passwordInput = getByPlaceholderText('Please Enter Password');

    fireEvent.changeText(emailInput, 'nayan@gmail.com');
    fireEvent.changeText(passwordInput, 'nayan@gmail.com');

    expect(emailInput.props.value).toBe('nayan@gmail.com');
    expect(passwordInput.props.value).toBe('nayan@gmail.com');
  });

  it('calls setAuth when login button is pressed', () => {
    const { getByText } = renderLoginScreen();

    const loginButton = getByText('Login');

    fireEvent.press(loginButton);

    expect(mockAuthContext.setAuth).toHaveBeenCalledTimes(1);
    expect(mockAuthContext.setAuth).toHaveBeenCalledWith(true);
  });
});
