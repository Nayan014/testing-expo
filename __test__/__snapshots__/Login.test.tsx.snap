// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen render and match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    height={100}
    source={
      {
        "uri": "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/1200px-React-icon.svg.png",
      }
    }
    testID="logo"
    width={110}
  />
  <View
    style={
      {
        "marginTop": 40,
        "width": "80%",
      }
    }
  >
    <TextInput
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Please Enter Email"
      style={
        {
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 0.5,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Please Enter Password"
      secureTextEntry={true}
      style={
        [
          {
            "borderColor": "gray",
            "borderRadius": 5,
            "borderWidth": 0.5,
          },
          {
            "marginBottom": 18,
            "marginTop": 14,
          },
        ]
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
